lib/cartopy/tests/test_img_transform.py

```diff
@@ -108,3 +108,20 @@ def test_gridding_data_outside_projection():
     assert_array_equal([-180, 180, -90, 90], extent)
     assert_array_equal(expected, image)
     assert_array_equal(expected_mask, image.mask)
+
+
+@pytest.mark.parametrize("target_prj", (ccrs.Mollweide(), ccrs.LambertAzimuthalEqualArea()))
+def test_invalid_regridding_with_invalid_extent(target_prj):
```

Comment:
these tests currently fail when running on the `main` branch with the error in #2199 and pass on the current branch. wasn't entirely sure what to check in terms of values though, so at present this is just checking that the transformation does not error.

Reply:
Note: on running tests locally, the following tests fail with scipy 1.11.1 on main but pass on this branch:

```
============================================================================================================== short test summary info ==============================================================================================================
FAILED lib/cartopy/tests/mpl/test_images.py::test_imshow - ValueError: 'x' must be finite, check for nan or inf values
FAILED lib/cartopy/tests/mpl/test_images.py::test_imshow_rgba_alpha - ValueError: 'x' must be finite, check for nan or inf values
FAILED lib/cartopy/tests/mpl/test_images.py::test_stock_img - ValueError: 'x' must be finite, check for nan or inf values
FAILED lib/cartopy/tests/mpl/test_images.py::test_pil_Image - ValueError: 'x' must be finite, check for nan or inf values
FAILED lib/cartopy/tests/mpl/test_images.py::test_background_img - ValueError: 'x' must be finite, check for nan or inf values
```
I think these tests are likely passing the CI on other PR's because the test suite spinup installs both scipy and pykdtree, and pykdtree takes precedence when deciding on a kdtree method.